sys/include/net/ng_netreg.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    ng_netreg  Network protocol registry
+ * @ingroup     net
+ * @{
+ *
+ * @file
+ * @brief   Definitions to register network protocol PIDs to use with
+ *          @ref net_netapi.
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+#ifndef NETREG_H_
+#define NETREG_H_
+
+#include <inttypes.h>
+
+#include "kernel_types.h"
+#include "net/ng_nettype.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Number of maximum entries to @ref ng_netreg
+ * @def NG_NETREG_SIZE
+ */
+#ifndef NG_NETREG_SIZE
+#define NG_NETREG_SIZE (NG_NETTYPE_NUMOF)
+#endif
+
+/**
+ * @brief   Entry to the @ref ng_netreg
+ */
+typedef struct ng_netreg_entry {
+    struct ng_netreg_entry *next;   /**< next element in list */
+
+    /**
+     * @brief   The demultiplexing context for the registering thread.
+     *
+     * @details This can be defined by the network protocol themselves.
+     *          E. g. protocol numbers / next header numbers in IPv4/IPv6,
+     *          ports in UDP/TCP, or similar.
+     */
+    uint16_t demux_ctx;
```

Comment:
This type does not exist in the most current version of netapi, since we moved all of the registration to this module. `netapi_reg_demux_ctx_t` is quite a mouthful compared to a simple `uint16_t`, so I left it out ;)